pkg/splunk/enterprise/configuration.go

```diff
@@ -926,6 +930,9 @@ func createOrUpdateAppUpdateConfigMap(client splcommon.ControllerClient, cr splc
 	// prepare the configMap data OR
 	// initialize the configMap data for this CR type,
 	// if it did not exist before
+	if crKindMap == nil {
```

Comment:
With the serialized access, is this code still needed? Or can we revert to creation at top of the method?

Reply:
We could revert it, but IMO think this is fine too!

Reply:
Hmm..with this code, under what circumstances do we expect `crKindMap` be nil and need a new map created, especially considering line 924? One scenario I can think of is when `splctrl.GetConfigMap(client, namespacedName)` fails, but do we even want to proceed in that case?

Reply:
This is a good question. I think it depends on how the json marshal takes care. At least, based on the original observations from this issue, seems like it never initializes the map(?), when the DB data is nil.

Reply:
Well, original code had `crKindMap := make(map[string]string)` in the beginning of the function - if whatever reason the `crKindMap == nil ` succeeds post this, the map can still be nil post this new code to make the map. I think we need to understand better why `crKindMap` became nil in the original code

Reply:
Ok, I think I discussed this before also but let me refresh again. The original problem happened when a previous CR(of same kind in same namespace) was still getting terminated while a new one is coming up. There is a small window, where while deleting the previous CR(and if it is the last CR to be deleted), we remove its entry from the configMap BUT the configMap itself is not deleted yet by the garbage collector of K8s controller. In this case, the data section of the configMap is nil. And now if a new CR is coming up, we will go to line 910 and we will not return `err` since we have a configMap but as soon as we reach line 923, since `configMap.Data` = nil, `crKindMap is also nil and hence we would crash. With the current fix, we are covering that scenario.
Also, as far as line 926 is concerned, `numOfObjects` will be 0 anyways so we are good there too.
I hope this clears the exact scenario and how this fix addresses that problem.